woocommerce/woocommerce-gutenberg-products-block
# ensure cart totals update when items are removed or quantity changed

Fixes #1830

~~This PR modifies the `DELETE /cart/items/:key` endpoint so that it always returns a complete cart object (previously returns no content).~~

This PR moves the `DELETE /cart/items/:key` endpoint to cart controller as `POST /cart/remove-item/?key=` so that it always returns a complete cart object (previously returns no content). Also moves the API for updating item quantity to `POST /cart/update-item`. The client actions, API docs and PHPunit tests have been updated alongside this change. 

The [idea here is that any endpoint that modifies the cart returns full cart](https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/1830#issuecomment-592469114), as other data may change (e.g. add a coupon => totals change, remove an item => totals change, shipping may change etc). 

Note that the only two APIs that have been moved are for deleting and removing cart items. I've moved these specifically because we need them to return full cart for the cart block to update totals (etc). There are other routes which could be moved, but we can discuss / do that as follow up - see related issue #1849.


### How to test the changes in this Pull Request:

1. Add a cart block to a page and publish.
2. Add some stuff to cart.
3. View cart page and remove items – totals and other cart details should be correct.
4. View cart page and change item quantities - totals and other details should update as items change.
